### DISCUSSION TITLE
Notifications GraphQL API

### DISCUSSION THREAD

**Discussion by tngranados** [community/community#13056] [Apps, API and Webhooks]
User notifications are not currently available in the GraphQL API. It's been [proposed earlier](https://github.community/t/get-notification-list-via-graphql-api/13836/12) to add them to the GraphQL API, but looks like it went nowhere. I think it would be pretty useful to have as current REST API is not very flexible.

For example, I was trying to get the list of notifications and whether I'm currently subscribed to the threads and with the current REST API I'd have to first, query for the list of notifications, then make query to each one to get the thread subscription.

**Comment by mobilenetworkltd** [comment]
Any progress on the GraphQL notifications ?

**Comment by harrysolovay** [comment]
This seems like a crucial feature for any GitHub apps that want to intelligently integrate with the notification system. Hoping a GH team member can weigh in.

**Comment by ddzy** [comment]
guys，any progress？

**Reply by kelvinpraises** [reply]
I checked the API documentation today, and unfortunately, I couldn't find any relevant information. I even downloaded the public schema and searched for a notifications object, but came up empty-handed. I just stumbled upon this thread now, so no luck so far.

**Comment by ranma2913** [comment]
I'd like to request for this too. It's pretty annoying to burn up my core rate limit on notification api calls when I could leverage one of the other rate limits like graphql.

**Comment by 0xdevalias** [comment]
> Seems like the Mobile app is using some graphQL API endpoints behind a feature flag. Would be great to implement those in `gh` as well 
> 
> _Originally posted by @pwntester in [#659](https://github.com/cli/cli/issues/659#issuecomment-1486436927)_

**Reply by 0xdevalias** [reply]
I also did some exploratory work on inspecting how GitHub Mobile's GraphQL API calls are working in https://github.com/cli/cli/issues/659#issuecomment-2599592102

Which I also documented on this gist for posterity:

- https://gist.github.com/0xdevalias/04ec31b12edaae727cd87d379499c14c#github-ios-graphql-query

**Comment by Fryuni** [comment]
So we had this feature implemented, entirely working, and documented. All for a grand total of four days and then it was removed?!?! Why?!

They were added to the API on [Jan 10](https://docs.github.com/en/graphql/overview/changelog#schema-changes-for-2025-01-10-1) and removed on [Jan 14](https://docs.github.com/en/graphql/overview/changelog#schema-changes-for-2025-01-14-1). And even tho they documented the removal on Jan 14 they continued to work (perfectly, had literally zero bugs with it) until this week when it was actually removed.

GH literally already had id done. What is the excuse? I mean besides intentionally providing a worse experience, which seems to be the motto of the changes recently.
